src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java

```diff
@@ -0,0 +1,196 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.service.archive.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.service.archive.controller.ArchiveController;
+import org.commonjava.indy.service.archive.model.dto.HistoricalContentDTO;
+import org.commonjava.indy.service.archive.util.HistoricalContentListReader;
+import org.commonjava.indy.service.archive.util.TransferStreamingOutput;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static javax.ws.rs.core.Response.noContent;
+
+@Path( "/api/archive" )
+public class ArchiveManageResources
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    HistoricalContentListReader reader;
+
+    @Inject
+    ArchiveController controller;
+
+    @Inject
+    ResponseHelper responseHelper;
+
+    @Operation( description = "Generate archive based on tracked content" )
+    @APIResponse( responseCode = "201", description = "The archive is created successfully" )
+    @RequestBody( description = "The tracked content definition JSON", name = "body", required = true,
+            content = @Content( schema = @Schema( implementation = HistoricalContentDTO.class ) ) )
+    @POST
+    @Path( "generate" )
+    @Consumes( APPLICATION_JSON )
+    public Response create( final @Context UriInfo uriInfo, final @Context HttpRequest request )
+    {
+        HistoricalContentDTO content;
+        try
+        {
+            String json = IOUtils.toString( request.getInputStream(), Charset.defaultCharset() );
```

Comment:
I think we may need to save json somewhere

Reply:
It's filed in the archive.